sys/suit/handlers_command_seq.c

```diff
@@ -207,34 +214,50 @@ static int _dtv_set_param(suit_manifest_t *manifest, int key,
 
     nanocbor_enter_map(it, &map);
 
+    suit_component_t *comp = _get_component(manifest);
+
     while (!nanocbor_at_end(&map)) {
         /* map points to the key of the param */
         int32_t param_key;
-        nanocbor_get_int32(&map, &param_key);
-        LOG_DEBUG("Current component index: %" PRIi32 "\n",
-                  manifest->component_current);
+        if (nanocbor_get_int32(&map, &param_key) < 0) {
+            return SUIT_ERR_INVALID_MANIFEST;
+        }
         LOG_DEBUG("param_key=%" PRIi32 "\n", param_key);
-        int res;
+        unsigned int type = nanocbor_get_type(&map);
+        /* Filter 'complex' types and only allow int, nint, bstr and tstr types
+         * for parameter values */
+        if (type > NANOCBOR_TYPE_TSTR) {
+            return SUIT_ERR_INVALID_MANIFEST;
+        }
+        suit_param_ref_t *ref;
         switch (param_key) {
-            case 6: /* SUIT URI LIST */
-                res = _param_get_uri_list(manifest, &map);
+            case 1: /* SUIT VENDOR ID */
+                ref = &comp->param_vendor_id;
                 break;
-            case 11: /* SUIT DIGEST */
-                res = _param_get_digest(manifest, &map);
+            case 2: /* SUIT URI LIST */
+                ref = &comp->param_class_id;
                 break;
-            case 12: /* SUIT IMAGE SIZE */
-                res = _param_get_img_size(manifest, &map);
+            case 3: /* SUIT DIGEST */
```

Comment:
Are there no defines/enum for that?

Reply:
No, there are defines for this, should have used those from the start :blush: